src/Microsoft.AspNet.OData.Shared/Formatter/Serialization/ODataResourceSerializer.cs

```diff
@@ -755,15 +770,15 @@ private void WriteDynamicComplexProperty(object propertyValue, IEdmTypeReference
                 if (navigationLink != null)
                 {
                     writer.WriteStart(navigationLink);
-                    WriteComplexAndExpandedNavigationProperty(navPropertyToExpand.Key, navPropertyToExpand.Value, resourceContext, writer);
+                    WriteComplexAndExpandedNavigationProperty(navPropertyToExpand.Key, navPropertyToExpand.Value, resourceContext, writer, null);
                     writer.WriteEnd();
                 }
             }
         }
 
         private void WriteComplexAndExpandedNavigationProperty(IEdmProperty edmProperty, SelectExpandClause selectExpandClause,
             ResourceContext resourceContext,
-            ODataWriter writer)
+            ODataWriter writer, IList<ExpandedNavigationSelectItem> selectExpandItems)
```

Comment:
> IList<ExpandedNavigationSelectItem> selectExpandItems [](start = 31, length = 54)

we can't get it from "selectExpandClause" ?

Reply:
I don't think so. We need the query options inside the expand as well.

Reply:
No. 

---
In reply to: [258238418](https://github.com/OData/WebApi/pull/1738#discussion_r258238418) [](ancestors = 258238418)